instrumentation/aws_sdk/lib/opentelemetry/instrumentation/aws_sdk/handler.rb

```diff
@@ -9,28 +9,37 @@ module Instrumentation
     module AwsSdk
       # Generates Spans for all interactions with AwsSdk
       class Handler < Seahorse::Client::Handler
-        def call(context) # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
+        SQS_SEND_MESSAGE = 'SQS.SendMessage'
+        SQS_SEND_MESSAGE_BATCH = 'SQS.SendMessageBatch'
+        SQS_RECEIVE_MESSAGE = 'SQS.ReceiveMessage'
+        SNS_PUBLISH = 'SNS.Publish'
+
+        def call(context) # rubocop:disable Metrics/AbcSize, Metrics/MethodLength, Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
           return super unless context
 
-          service_name = context.client.class.api.metadata['serviceId'] || context.client.class.to_s.split('::')[1]
+          service_name = context.client.class.to_s.split('::')[1]
           operation = context.operation&.name
+          client_method = "#{service_name}.#{operation}"
           attributes = {
             'aws.region' => context.config.region,
             OpenTelemetry::SemanticConventions::Trace::RPC_SYSTEM => 'aws-api',
             OpenTelemetry::SemanticConventions::Trace::RPC_METHOD => operation,
             OpenTelemetry::SemanticConventions::Trace::RPC_SERVICE => service_name
           }
           attributes[SemanticConventions::Trace::DB_SYSTEM] = 'dynamodb' if service_name == 'DynamoDB'
+          MessagingHelper.apply_sqs_attributes(attributes, context, client_method) if service_name == 'SQS'
+          MessagingHelper.apply_sns_attributes(attributes, context, client_method) if service_name == 'SNS'
 
-          tracer.in_span("#{service_name}.#{operation}", attributes: attributes, kind: OpenTelemetry::Trace::SpanKind::CLIENT) do |span|
+          tracer.in_span(span_name(context, client_method), attributes: attributes, kind: span_kind(operation)) do |span|
```

Comment:
I don't think the cardinality of `span_name()` is appropriate for messaging spans., Below are the  `span_name` and `MessagingHelper.queue_name` methods. 

Specifically, it seems possible to me that, SNS will result in incredibly high cardinality span names with a phone number in the span name, like for example, `867 5309 send`. However, the messaging span conventions on span name (and general span name conventions also apply here) [state](https://github.com/open-telemetry/opentelemetry-specification/blob/a1a8676a43dce6a4e447f65518aef8e98784306c/specification/trace/semantic_conventions/messaging.md#span-name):

> The destination name SHOULD only be used for the span name if it is known to be of low cardinality (cf. general span name guidelines). This can be assumed if it is statically derived from application code or configuration. Wherever possible, the real destination names after resolving logical or aliased names SHOULD be used. If the destination name is dynamic, such as a conversation ID or a value obtained from a Reply-To header, it SHOULD NOT be used for the span name. In these cases, an artificial destination name that best expresses the destination, or a generic, static fallback like "(temporary)" for temporary destinations SHOULD be used instead.

Additionally, having something with obvious PII implications in the span name feels gross to me. I realize this is how Node does it currently but, frankly, I think that's incorrect and i'd prefer to see this pattern not extended into Ruby. Or, we should get some clarification from the AWS team on how to represent this.

For now i'd prefer to see Phone number replaced with something either a message attribute of much lower cardinality, or something static like `phone_number send`

```ruby
        def span_name(context, client_method)
          case client_method
          when SQS_SEND_MESSAGE, SQS_SEND_MESSAGE_BATCH, SNS_PUBLISH
            "#{MessagingHelper.queue_name(context)} send"
          when SQS_RECEIVE_MESSAGE
            "#{MessagingHelper.queue_name(context)} receive"
          else
            client_method
          end
        end
```

```ruby
          def queue_name(context) # rubocop:disable Metrics/CyclomaticComplexity
            topic_arn = context.params[:topic_arn]
            target_arn = context.params[:target_arn]
            phone_number = context.params[:phone_number]
            queue_url = context.params[:queue_url]

            if topic_arn || target_arn
              arn = topic_arn || target_arn
              begin
                return arn.split(':')[-1]
              rescue StandardError
                return arn
              end
            end

            return phone_number if phone_number

            return queue_url.split('/')[-1] if queue_url

            'unknown'
          end
```

Reply:
updated 👍
https://github.com/open-telemetry/opentelemetry-ruby/pull/1026/commits/b28178ef1a1677d6d039b5449fa9a35a7febe34b